### ISSUE TITLE
Authenticator.keychain fails while saving multiple times the same instagram secret

### ISSUE THREAD

**Issue by biox86** [sbertix/Swiftagram#208]
**Dependencies version**
Swiftagram 5.1.0
Composable request 5.0.9
SwCrypt 5.1.4
Swiftchain 1.0.5

**Bug description**
If you have already saved a secret with Authenticator.keychain the second time will respond with a failure:
Swiftchain.Keychain.Error.status(-25303) 

**Steps to reproduce**
Login with a webview saving your secret with Authenticator.keychain. Repeat the process a second time to get the failure.

**Example code snippet**
import Foundation
import UIKit
import ComposableRequest
import Swiftagram

class LoginViewController: UIViewController {
    var completion: ((Secret) -> Void)? {
        didSet {
            guard oldValue == nil, let completion = completion else { return }
            DispatchQueue.main.asyncAfter(deadline: .now()) {
                Authenticator.keychain
                    .visual(filling: self.view)
                    .authenticate()
                    .receive(on: RunLoop.main)
                    .sink(receiveCompletion: { print($0); self.dismiss(animated: true, completion: nil) },
                          receiveValue: completion)
                    .store(in: &self.bin)
            }
        }
    }
    private var bin: Set<AnyCancellable> = []
}
class viewController: UIViewController
{
    override func viewDidLoad()
    {
        super.viewDidLoad()
        let controller = LoginViewController()
        controller.completion = { secret in
            print("Secret saved ", secret.identifier)
        }
        self.present(controller, animated: true, completion: nil)
    }
}

**Expected behavior**
The first login attempt will correctly print the secret and the "Secret saved" log. The next tries will end up in failure printing Swiftchain.Keychain.Error.status(-25303) 
Resetting the device and device data will end up in solving the problem just for the first attempt and loop the same failure again.

**Additional context**
I am opening this bug in referral to https://github.com/sbertix/Swiftagram/discussions/182
Every dependency is are already up to date.

**Device and system**
Xcode Version 12.5 (12E262)
Simulator iPhone 12 IOS 14.5
Device iPhone 12 IOS 14.5.1

**Comment by sbertix** [comment]
So basically this happens when you try to log in an already persisted account, correct? 

When, instead of fetching the persisted `Secret`, you present the login sheet again, when trying to overwrite the previous one the exception gets raised, correct?

**Comment by biox86** [comment]
Exactly, I know it doesn't make much sense to re-log the same account once you already have the secret and I don't do that either honestly, but I noticed this situation while doing some tests.
I guess it should be better to overwrite the new value correctly or maybe returning the already cached one.

**Comment by sbertix** [comment]
Yep, I was just trying to make sure, cause it's harder to test the new storage system for these edge cases. 
I'll work on it and push a fix in the next 24 hours, hopefully. 
Thanks for pointing this out man 💪

**Comment by biox86** [comment]
No problem, thank you for being this responsive!

**Comment by sbertix** [comment]
Okay, so apparently this is an issue with **Swiftchain** and unexpected keychain attributes. 
This is not related to `5.*` per se, simply, before `5.*`, storages would never `throw`, but `Secret`s still wouldn't be updated. So it's definitely a good thing that this made it so it was actually catchable. 
I'm gonna tinker a bit with keychain implementation again, and see what I can come out with.

**Comment by biox86** [comment]
Nice to know that this bug pointed out that problem then. Thanks! If you need some feedback for testing stuff or something, just let me know.
